xbmc/cores/VideoPlayer/VideoRenderers/DebugRenderer.cpp

```diff
@@ -6,178 +6,187 @@
 {
+  if (!m_isConfigured)
+    return;
+
+  // FIXME: We currently force ASS_Event's and the current PTS
+  // of rendering with fixed values to allow perpetual
+  // display of on-screen text. It would be appropriate for Libass
+  // provide a way to allow fixed on-screen text display
+  // without use all these fixed values.
+
   m_overlayRenderer.Release(0);
 
   if (info.audio != m_strDebug[0])
   {
     m_strDebug[0] = info.audio;
-    if (m_overlay[0])
-      m_overlay[0]->Release();
-    m_overlay[0] = new CDVDOverlayText();
-    m_overlay[0]->AddElement(new CDVDOverlayText::CElementText(m_strDebug[0]));
+    m_adapter->AddSubtitle(m_strDebug[0].c_str(), 0., 5000000.);
   }
   if (info.video != m_strDebug[1])
   {
     m_strDebug[1] = info.video;
-    if (m_overlay[1])
-      m_overlay[1]->Release();
-    m_overlay[1] = new CDVDOverlayText();
-    m_overlay[1]->AddElement(new CDVDOverlayText::CElementText(m_strDebug[1]));
+    m_adapter->AddSubtitle(m_strDebug[1].c_str(), 0., 5000000.);
   }
   if (info.player != m_strDebug[2])
   {
     m_strDebug[2] = info.player;
-    if (m_overlay[2])
-      m_overlay[2]->Release();
-    m_overlay[2] = new CDVDOverlayText();
-    m_overlay[2]->AddElement(new CDVDOverlayText::CElementText(m_strDebug[2]));
+    m_adapter->AddSubtitle(m_strDebug[2].c_str(), 0., 5000000.);
   }
   if (info.vsync != m_strDebug[3])
   {
     m_strDebug[3] = info.vsync;
-    if (m_overlay[3])
-      m_overlay[3]->Release();
-    m_overlay[3] = new CDVDOverlayText();
-    m_overlay[3]->AddElement(new CDVDOverlayText::CElementText(m_strDebug[3]));
+    m_adapter->AddSubtitle(m_strDebug[3].c_str(), 0., 5000000.);
   }
 
-  for (int i = 0; i < 4; i++)
-    m_overlayRenderer.AddOverlay(m_overlay[i], 0, 0);
+  auto overlay = m_adapter->CreateOverlay();
+  m_overlayRenderer.AddOverlay(overlay, 1000000., 0);
 }
 
 void CDebugRenderer::SetInfo(DEBUG_INFO_VIDEO& video, DEBUG_INFO_RENDER& render)
 {
+  if (!m_isConfigured)
+    return;
+
+  // FIXME: We currently force ASS_Event's and the current PTS
+  // of rendering with fixed values to allow perpetual
+  // display of on-screen text. It would be appropriate for Libass
+  // provide a way to allow fixed on-screen text display
+  // without use all these fixed values.
+
   m_overlayRenderer.Release(0);
 
   if (video.videoSource != m_strDebug[0])
   {
     m_strDebug[0] = video.videoSource;
-    if (m_overlay[0])
-      m_overlay[0]->Release();
-    m_overlay[0] = new CDVDOverlayText();
-    m_overlay[0]->AddElement(new CDVDOverlayText::CElementText(m_strDebug[0]));
+    m_adapter->AddSubtitle(m_strDebug[0].c_str(), 0., 5000000.);
   }
   if (video.metaPrim != m_strDebug[1])
   {
     m_strDebug[1] = video.metaPrim;
-    if (m_overlay[1])
-      m_overlay[1]->Release();
-    m_overlay[1] = new CDVDOverlayText();
-    m_overlay[1]->AddElement(new CDVDOverlayText::CElementText(m_strDebug[1]));
+    m_adapter->AddSubtitle(m_strDebug[1].c_str(), 0., 5000000.);
```

Comment:
You have a few magic numbers here. What happens when the end time is reached? The ass "spec" does not preclude "always visible" overlays, so we need to ensure the overlay time is extended, or am I understanding the current logic wrong?

Reply:
This is what I tried to explain to you on slack,
here you don't have to rely on start/duration times, but only to the actual PTS  passed to the libass renderer
from this line we pass "1000000." so we have always the current pts to 1sec
https://github.com/CastagnaIT/xbmc/blob/subtitles_libass/xbmc/cores/VideoPlayer/VideoRenderers/DebugRenderer.cpp#L84
in this way we have "always visible" events because the libass renderer elaborate every time the images for the PTS position of 1 sec

Reply:
now that i see seem create too much overlays over the time, i am not sure, i will recheck